stock_split_picking_kit/tests/test_stock_split_picking_kit.py

```diff
@@ -0,0 +1,245 @@
+                "purchase_ok": False,
+            }
+        )
+        cls.bom_model = cls.env["mrp.bom"]
+        cls.bom_garden_table = cls.bom_model.create(
+            {
+                "product_tmpl_id": cls.tmpl_garden_table.id,
+                "product_id": cls.product_garden_table.id,
+                "type": "phantom",
+                "bom_line_ids": [
+                    (
+                        0,
+                        0,
+                        {
+                            "product_id": cls.product_garden_table_leg.id,
+                            "product_qty": 4.0,
+                        },
+                    ),
+                    (
+                        0,
+                        0,
+                        {
+                            "product_id": cls.product_garden_table_top.id,
+                            "product_qty": 1.0,
+                        },
+                    ),
+                ],
+            }
+        )
+        # cls.bom_garden_furniture = cls.bom_model.create(
+        #     {
+        #         "product_tmpl_id": cls.tmpl_garden_furniture.id,
+        #         "product_id": cls.product_garden_furniture.id,
+        #         "type": "phantom",
+        #         "bom_line_ids": [
+        #             (
+        #                 0,
+        #                 0,
+        #                 {"product_id": cls.product_garden_chair.id, "product_qty": 4.0},
+        #             ),
+        #             (
+        #                 0,
+        #                 0,
+        #                 {"product_id": cls.product_garden_table.id, "product_qty": 1.0},
+        #             ),
+        #         ],
+        #     }
+        # )
+
+    @classmethod
+    def _create_picking(cls, lines):
+        picking_form = Form(cls.env["stock.picking"])
+        picking_form.picking_type_id = cls.picking_type
+        picking_form.partner_id = cls.partner
+        for product, qty in lines:
+            with picking_form.move_ids_without_package.new() as move:
+                move.product_id = product
+                move.product_uom_qty = qty
+        picking = picking_form.save()
+        picking.action_confirm()
+        return picking
+
+    def _get_picking_ids_from_action(self, res, expected_quantity):
+        """Return the new picking found in the action returned by the wizard."""
+        id_list = res["domain"][0][2]
+        self.assertEqual(len(id_list), expected_quantity)
+        return self.env["stock.picking"].browse(id_list)
+
+    @classmethod
+    def _get_kit_quantity(cls, picking, bom):
+        """Returns the quantity of kits in a transfer."""
+        filters = {
+            "incoming_moves": lambda m: m.location_id.usage == "supplier",
+            "outgoing_moves": lambda m: m.location_id.usage != "supplier",
+        }
+        kit_quantity = picking.move_lines._compute_kit_quantities(
+            bom.product_id, 100, bom, filters
```

Comment:
Maybe like you did above
```suggestion
            bom.product_id, max(picking.move_lines.mapped("product_qty")), bom, filters
```